lib/PuppeteerSharp/NetworkManager.cs

```diff
@@ -23,7 +23,7 @@ internal class NetworkManager
         private readonly ILogger _logger;
         private Dictionary<string, string> _extraHTTPHeaders;
         private Credentials _credentials;
-        private readonly List<string> _attemptedAuthentications = new List<string>();
+        private readonly ConcurrentSet<string> _attemptedAuthentications = new ConcurrentSet<string>();
```

Comment:
`ConcurrentBag` won't help?

Reply:
`Contains` for `List` and `ConcurrentBag` is O(n) while for `ConcurrentDictionary` it's O(1).
It would work too though. Would you prefer to declare it as `ConcurrentDictionary` directly?
I see no good reason to use `ConcurrentBag` over `ConcurrentDictionary`.